File: Comparision/MSDIAL_data/DeepDIA-process.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jan 17 10:55:44 2020

@author: hcji
"""

import pandas as pd
from DeepDIA.DeepDIA import DeepDIA_process

f = 'E:/project/MSDIAL_data/20140809_MSDIAL_DemoFiles_Swath (wiff.wiffscan)/Nega_Swath_QC_1_1.mzML'
features = pd.read_csv('E:/project/MSDIAL_data/20140809_MSDIAL_DemoFiles_Swath (wiff.wiffscan)/Nega_Swath_QC_1_1.feature.csv')
features = features[['mz', 'rt', 'maxo']]
features.columns = ['mz', 'rt', 'intensity']
dia_result = DeepDIA_process(f, features)
dia_result.to_csv('Comparision/MSDIAL_data/results/DeepDIA_results/Nega_Swath_QC_1_1.ms2.csv')

File: Comparision/MSDIAL_data/DeepDIA-process.R
# xcms version 3.8.1
library(xcms)

get_ms1_features <- function(file, snthresh = 10, prefilter = c(1, 100), noise = 100, ppm = 30, peakwidth = c(5, 60)){
  data <- readMSData(file, mode = "onDisk") 
  cwp <- CentWaveParam(snthresh = snthresh, noise = noise, prefilter = prefilter, ppm = ppm, peakwidth = peakwidth)
  data <- findChromPeaks(data, param = cwp)
  peaks <- chromPeaks(data)
  return(peaks)
}

file <- 'E:/project/MSDIAL_data/20140809_MSDIAL_DemoFiles_Swath (wiff.wiffscan)/Nega_Swath_QC_1_1.mzML'
features = get_ms1_features(file, snthresh = 5, prefilter = c(1, 1000))
write.csv(features, stringr::str_replace(file, '.mzML', '.feature.csv'))